### DISCUSSION TITLE
Possibility to skip indexing large files?

### DISCUSSION THREAD

**Discussion by edvreu** [erlang-ls/erlang_ls#1248] [General]
I am working with a repo that has huge .erl files with generated protocol code (up to 10 megabytes or 100 000 lines of code).
Since it is generated code I rarely navigate in these files, so indexing them just wastes time (at least 20 min) and memory.
Would it be possible to add a size limit (2 MB?) for files that will be indexed?

**Comment by edvreu** [comment]
Support was added for this shortly after my question about it. I can't find anything in the configuration documentation, but add this to the `erlang_ls.config` to get it working:
```
indexing:
  skip_generated_files: true
  generated_files_tag: "Generated"
```

**Reply by robertoaloi** [reply]
The reason why this is undocumented is that, shortly after implementing the _skip_ functionality, we switched indexing to an on-demand model, which should make the _skipping_ not so relevant anymore.

**Reply by edvreu** [reply]
Ok, I still feel it makes a difference (just by looking at how much memory the erlang_ls process uses) so please keep the functionality.
